File: hmi/sdk/Linux/cpp/src/api/safety.cpp
// Copyright (c) 2026 WeGo Robotics. All rights reserved.
// SPDX-License-Identifier: LicenseRef-Wego-Proprietary

#include "shalom/api.hpp"

namespace shalom {

RobotApi::RobotApi(Client &client) : client_(client) {}

std::string RobotApi::emergencyStop(std::string *err)
{
    return client_.sendRequest("cmd/estop", "{}", err);
}

std::string RobotApi::releaseEmergencyStop(std::string *err)
{
    return client_.sendRequest("cmd/estop_release", "{}", err);
}

std::string RobotApi::setMode(const std::string &mode, std::string *err)
{
    if (mode != "auto" && mode != "manual") {
        if (err)
            *err = "mode must be auto or manual";
        return {};
    }
    return client_.sendRequest("cmd/mode", "{\"mode\":\"" + mode + "\"}", err);
}

std::string RobotApi::request(const std::string &channel, const std::string &payloadJson,
                              std::string *err)
{
    return client_.sendRequest(channel, payloadJson, err);
}

}  // namespace shalom

File: hmi/sdk/Linux/python/shalom_sdk/api/safety.py
# Copyright (c) 2026 WeGo Robotics. All rights reserved.
# SPDX-License-Identifier: LicenseRef-Wego-Proprietary

"""Emergency-stop and operating-mode commands."""

from __future__ import annotations

from ..types import Response
from ._base import _ApiBase


class SafetyApi(_ApiBase):
    def emergency_stop(self, timeout_s: float = 3.0) -> Response:
        return self._client.request("cmd/estop", timeout_s=timeout_s)

    def release_emergency_stop(self, timeout_s: float = 3.0) -> Response:
        return self._client.request("cmd/estop_release", timeout_s=timeout_s)

    def set_mode(self, mode: str, timeout_s: float = 3.0) -> Response:
        if mode not in {"auto", "manual"}:
            raise ValueError("mode must be 'auto' or 'manual'")
        return self._client.request("cmd/mode", {"mode": mode}, timeout_s)
